### PR TITLE
Added ELBs on AWS for ingress and master nodes.

### PR DESCRIPTION
**Pull request by emmetthitz**
Only minor thing I'm going to leave out of this is the fact that they don't log (yet).

### CODE REVIEW

File: pkg/provision/aws.go [hunk 1]
```diff
@@ -120,8 +120,17 @@ func (aws *AWS) buildPopulatedPlan(plan install.Plan) (*install.Plan, error) {
...
-	mng.LoadBalancedShortName = tfNodes.IPs[0]
+	if masterNodes.ExpectedCount > 1 {
+		mlb, err := aws.getLoadBalancer(plan.Cluster.Name, "master")
+		if err != nil {
+			return nil, err
+		}
+		mng.LoadBalancedFQDN = mlb
+		mng.LoadBalancedShortName = mlb
+	} else {
+		mng.LoadBalancedFQDN = mng.Nodes[0].InternalIP
```
Comment (emmetthitz) [inline pkg/provision/aws.go]: oh oops.

File: terraform/providers/aws/main.tf [hunk 2]
```diff
@@ -4,9 +4,9 @@ provider "aws" {
   $ export AWS_SECRET_ACCESS_KEY=YOUR_AWS_SECRET_ACCESS_KEY
   $ export AWS_DEFAULT_REGION=us-east-1
   */
-  region = "${var.region}"
-  access_key = "${var.access_key}"
-  secret_key = "${var.secret_key}"
+  region      = "${var.region}"
```
Comment (emmetthitz) [inline terraform/providers/aws/main.tf]: It's probably worth breaking this whole thing up into its respective components so we don't have 500 lines of a monolith terraform script. TODO.

File: terraform/providers/aws/main.tf [hunk 3]
```diff
@@ -121,72 +120,313 @@ resource "aws_subnet" "kismatic_private" {
...
+    "kismatic/version"      = "${var.version}"
+    "kismatic/subnet"       = "ingress"
+    "kubernetes.io/cluster" = "${var.cluster_name}"
+  }
+  lifecycle {
+    ignore_changes = ["tags.kismatic/dateCreated", "tags.Owner", "tags.PrincipalID"]
+  }
+}
+
+resource "aws_security_group" "kismatic_public" {
```
Comment (alexbrand) [inline terraform/providers/aws/main.tf]: Can we call this `kismatic_ssh`, or something? I think it's going to make it more clear when referenced in other resources.

File: pkg/provision/provision.go [hunk 4]
```diff
@@ -49,6 +49,29 @@ type Provisioner interface {
...
+
+	//load balancer
+	tfCmdOutputLB := exec.Command(tf.BinaryPath, "output", "-json", tfOutLB)
+	tfCmdOutputLB.Dir = cmdDir
+	stdoutStderrLB, err := tfCmdOutputLB.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("Error collecting terraform output: %s", stdoutStderrLB)
+	}
+	lbData := tfOutputVar{}
+	json.Unmarshal(stdoutStderrLB, &lbData)
```
Comment (alexbrand) [inline pkg/provision/provision.go]: We need to handle the error here, as this can fail.

File: terraform/providers/aws/main.tf [hunk 5]
```diff
@@ -121,72 +120,313 @@ resource "aws_subnet" "kismatic_private" {
...
+
+  //access_logs {
+  //  bucket = "${aws_s3_bucket.lb_logs.bucket}"
+  //  bucket_prefix = "${var.cluster_name}/ingress"
+  //}
+
+  listener {
+    instance_port     = 6443
+    instance_protocol = "tcp"
+    lb_port           = 4443
```
Comment (alexbrand) [inline terraform/providers/aws/main.tf]: Typo

File: terraform/providers/aws/main.tf (hunk 3 above)
Comment (emmetthitz) [reply terraform/providers/aws/main.tf]: Yeah, sure.

**Comment by dkoshkin** [comment]
LGTM

**Pull request merged**